runtime/js/41_prompt.js

```diff
@@ -1,79 +1,142 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 const core = globalThis.Deno.core;
+const ops = core.ops;
 const primordials = globalThis.__bootstrap.primordials;
 import { isatty } from "ext:runtime/40_tty.js";
 import { stdin } from "ext:deno_io/12_io.js";
-const { ArrayPrototypePush, StringPrototypeCharCodeAt, Uint8Array } =
-  primordials;
-const LF = StringPrototypeCharCodeAt("\n", 0);
-const CR = StringPrototypeCharCodeAt("\r", 0);
+import { bold, cyan, italic, yellow } from "ext:deno_console/01_colors.js";
+const {
+  Array,
+  ArrayPrototypeFill,
+  ArrayPrototypeJoin,
+  ArrayPrototypeMap,
+  Error,
+  Map,
+  Uint8Array,
+  StringFromCodePoint,
+} = primordials;
+
+const LINE_UP = "\x1b[1A";
+const LINE_CLEAR = "\x1b[2K";
+const HIDE_CURSOR = "\x1b[?25l";
+const SHOW_CURSOR = "\x1b[?25h";
+const UP_ARROW = "\x1b[A";
+const DOWN_ARROW = "\x1b[B";
+const ESC = "\x1b";
+const CTRL_C = "\x03";
+const CTRL_D = "\x04";
 
 function alert(message = "Alert") {
   if (!isatty(stdin.rid)) {
-    return;
+    throw new Error("Cannot use `alert` in a non-interactive environment");
```

Comment:
Returning undefined here is aligned to what browsers do when dialog boxes are disabled. ref. https://github.com/denoland/deno/pull/7507#pullrequestreview-489404682

Reply:
Ahh OK - I got confused by an old version that was returning `null` for all the modals in non-interactive environments, which definitely seems wrong. I'll revert to return `undefined`, `null`, and `false` respectively.